wrappers/net/grpc/grpc_decorator.go

```diff
@@ -0,0 +1,87 @@
+package epsagongrpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/metadata"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+
+func createGRPCEvent(origin string, method string, eventID string) *protocol.Event {
+	errorcode := protocol.ErrorCode_OK
+
+	return &protocol.Event{
+		Id:        eventID + uuid.New().String(),
+		Origin:    origin,
+		StartTime: tracer.GetTimestamp(),
+		ErrorCode: errorcode,
+		Resource: &protocol.Resource{
+			Type:      "grpc",
+			Operation: method,
+			Metadata:  map[string]string{},
+		},
+	}
+}
+
+
+func decoratePostGRPCRunner(handlerWrapper *epsagon.GenericWrapper) {
+	runner := handlerWrapper.GetRunnerEvent()
+	if runner != nil {
+		runner.Resource.Type = "grpc"
+	}
+}
+
+
+func decorateGRPCRequest(Resource *protocol.Resource, ctx context.Context, fullMethod string, req interface{}) {
```

Comment:
would rename, maybe to extractFromGRPCRequest ? or something similar (as its not really decorate it but just extract information from it)

Reply:
fixed